docs/refs/crd-upgrade-safety.md

```diff
@@ -0,0 +1,218 @@
+# Custom Resource Definition Upgrade Safety
+
+When you update a Custom Resource Definition (CRD), OLM runs a CRD Upgrade Safety preflight
+check to ensure backwards compatibility with previous versions of that CRD. The CRD update
+must pass the validation checks before the change is allowed to progress on a cluster.
+
+## Prohibited CRD Upgrade Changes
+
+The following changes to an existing CRD will be caught by the CRD Upgrade
+Safety preflight check and prevent the upgrade:
+- The scope changes from Cluster to Namespace or from Namespace to Cluster
+- An existing stored version of the CRD is removed
+- A new required field is added to an existing version of the CRD
+- An existing field is removed from an existing version of the CRD
+- An existing field type is changed in an existing version of the CRD
+- A new default value is added to a field that did not previously have a default value
+- The default value of a field is changed
+- An existing default value of a field is removed
+- New enum restrictions are added to an existing field which did not previously have enum restrictions
+- Existing enum values from an existing field are removed
+- The minimum value of an existing field is increased in an existing version
+- The maximum value of an existing field is decreased in an existing version
+- Minimum or maximum field constraints are added to a field that did not previously have constraints
+
+!!! note
+    The rules for changes to minimum and maximum values apply to `minimum`, `minLength`,
+    `minProperties`, `minItems`, `maximum`, `maxLength`, `maxProperties`, and `maxItems` constrants.
+
+If the CRD Upgrade Safety preflight check encounters one of the disallowed upgrade changes,
+it will log an error for each disallowed change detected in the CRD upgrade.
+
+!!! tip
+    In cases where a change to the CRD does not fall into one of the disallowed change categories
+    but is also unable to be properly detected as allowed, the CRD Upgrade Safety preflight check
+    will prevent the upgrade and log an error for an "unknown change."
+
+If you identify any preflight checks that should be implemented to prevent issues during CRD upgrades, please [create a new issue](https://github.com/operator-framework/operator-controller/issues).
+
+
+## Allowed CRD Upgrade Changes
+
+The following changes to an existing CRD are safe for backwards compatibility and will
+not cause the CRD Upgrade Safety preflight check to halt the upgrade:
+- Adding new enum values to the list of allowed enum values in a field
```

Comment:
This list also needs an extra line break between the lead-in sentence and the list to fix the rendering in the preview:
![image](https://github.com/user-attachments/assets/1985d877-b43a-4e28-9a5e-25ad3fc6a4d7)